From b2f7634c9dbc0ef68640f0571787d92300e9f9f9 Mon Sep 17 00:00:00 2001
From: Stefan Strogin <stefan@steils.org>
Date: Sat, 15 Aug 2020 07:18:31 +0300
Subject: [PATCH] ec_sslwrap: fix compilation with LibreSSL

Disable taking over SNI extension from ClientHello and SSL configuration
operations until LibreSSL supports the required API.

Fixes: https://github.com/Ettercap/ettercap/issues/1068
Upstream-Status: Submitted
[https://github.com/Ettercap/ettercap/pull/1069]
Signed-off-by: Stefan Strogin <steils@gentoo.org>
---
 src/ec_sslwrap.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ec_sslwrap.c b/src/ec_sslwrap.c
index b9f26a14..1e4c24fc 100644
--- a/src/ec_sslwrap.c
+++ b/src/ec_sslwrap.c
@@ -71,11 +71,11 @@
 #define TLS_server_method SSLv23_server_method
 #endif
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && !defined(LIBRESSL_VERSION_NUMBER)
 #define HAVE_OPENSSL_1_1_0
 #endif
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10101000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10101000L) && !defined(LIBRESSL_VERSION_NUMBER)
 #define HAVE_OPENSSL_1_1_1
 #endif
 
-- 
2.28.0

